assets/src/js/Components/ArchivePicker/blockUpdateFunctions.js

```diff
@@ -39,3 +39,59 @@ export const updateHappyPointAttributes = id => {
   // Need to update only the id property for the Happy Point block
   return {attributes: {id}};
 };
+
+export const updateMediaAndTextAttributes = (image, currentBlock) => {
+  const div = document.createElement('div');
+
+  if (currentBlock.originalContent) {
+    div.innerHTML = currentBlock.originalContent;
+    div.querySelector('img').setAttribute('class', `wp-image-${image.id}`);
+    div.querySelector('img').src = image.source_url;
+  } else {
+    div.innerHTML = `
+    <div class="wp-block-media-text is-stacked-on-mobile">
+      <figure class="wp-block-media-text__media">
+        <img src=${image.source_url} alt=${image.alt_text} class="wp-image-${image.id} size-full"/>
+      </figure>
+      <div class="wp-block-media-text__content">
+      </div>
+    </div>
+    `;
+  }
+
+  const updatedAttributes = {
+    attributes: {
+      mediaLink: image.link,
+      mediaUrl: image.source_url,
+    },
+    originalContent: div.innerHTML,
+  };
+
+  return updatedAttributes;
+};
+
+export const updateCarouselBlockAttributes = (image, currentBlock) => {
+  const imageArray = currentBlock.attributes.slides;
+  const currentImageIndex = currentBlock.attributes.currentImageIndex;
+  const {thumbnail, medium_large, medium, large, full} = image.media_details.sizes;
+  const imageSrcSet = `
+  ${thumbnail.source_url} ${thumbnail.width}w,
+  ${medium_large.source_url} ${medium_large.width}w,
+  ${medium.source_url} ${medium.width}w,
+  ${large.source_url} ${large.width}w,
+  ${full.source_url} ${full.width}w,
+  `;
+
+  imageArray[currentImageIndex].image = image.id;
+  imageArray[currentImageIndex].image_url = image.source_url;
+  imageArray[currentImageIndex].image_srcset = imageSrcSet;
+
+  const updatedAttributes = {
+    attributes: {
+      slides: imageArray,
+    },
+  };
+
+  return updatedAttributes;
```

Comment:
We can return directly the `attributes` instead of defining a `const`

Reply:
This is for the dispatch method that updates the block, you have to use a certain structure to pass to the function.

Reply:
I think what Dan meant is that you could directly write
```
return {
  attributes: {
    slides,
  },
};
```
instead of creating the `updatedAttributes` constant